docs/tutorials/basics/basics_backend.md

```diff
@@ -0,0 +1,99 @@
+estimators, the following will come natural.
+[launch_sagemaker_backend.py](scripts/launch_sagemaker_backend.py) is a
+launcher script for MOBSTER (promotion-type), using the SageMaker instead of
+the local back-end.
+* [1] The only difference is that `backend`, feeding into the `Tuner`, is
+  different. Instead of the code `entry_point` only, you need to specify a
+  [SageMaker estimator](https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html)
+  to be used for executing your code as a SageMaker training job. Our running
+  example is in PyTorch, so we use the `PyTorch` estimator. Except for
+  `entry_point`, arguments depend on the estimator type. Some of the most
+  relevant ones are `instance_type`, `instance_count` (one training job can use
+  several instances), and `role`. The latter is your execution role for the
+  training job: with `get_execution_role()`, you use the default one associated
+  with your AWS account. Finally, `max_run` limits the time of a training job,
+  make sure this is not much shorter than your experiment time.
+  Apart from `sm_estimator`, you can also specify the names of metrics reported
+  to Syne Tune in `metrics_names`, these will be displayed in the dashboard of
+  the training job.
+* If your training script needs extra dependencies not contained in the
+  SageMaker framework, you need to list them in a file `requirements.txt` stored
+  in the same directory as your script. Our running example needs `filelock`,
+  for example.
+
+This launcher script shows how to run training code whose dependencies are
+captured by a SageMaker framework (`PyTorch` in this case), and extra ones
+are specified by a `dependencies.txt` file. However, you may prefer to use a
+Docker image for your training code.
+[launch_height_sagemaker_custom_image.py](examples/launch_height_sagemaker_custom_image.py)
+is an example for how to use such a custom image.
+
+
+## Properties of the SageMaker Back-end
+
+Our running example is not a good candidate to use the SageMaker back-end for.
+It is small, cheap and easy to run, and for such tuning problems, the local
+back-end is the preferred choice. The SageMaker back-end starts a new training
+job for every trial. While this offers great flexibility (you can select a
+SageMaker framework, instance type, instance count), it comes with a substantial
+start-up overhead (often almost two minutes). If training jobs are much longer
+than that, as is often the case with `FIFOScheduler`, these over-heads may not
+matter much.
+
+However, for early stopping schedulers like `HyperbandScheduler`, most trials
+do not run for long, but are stopped or paused after one or a few epochs. For such
+algorithms start-up overheads can matter a lot more. On the other hand, they 
+also profit from increased parallelization.
+
+SageMaker training jobs are not only slow to start, but also have quite some
+overhead for getting stopped. Therefore, if you use ASHA or MOBSTER with the
```

Comment:
Can you put more nuance to this sentence?

Reply:
In general, if something is slow, I will tell users it is slow. If we want to have any chance of getting adoption from external people, we should agree that being honest about weaknesses is a good strategy. And we also need to be clear about the differences between the back-ends. The SM back-end has strenghts and weaknesses, and once there is back-end number 3, then that one will have strengths and weaknesses compared to the SM back-end as well.

Reply:
I am not saying that we should not mention that there is a delay, nor that we should mention backend pros and cons. 

I am saying that the formulation "SageMaker training jobs are not only slow to start, but also have quite some overhead for getting stopped" has a quite negative stance that can be avoided or precised. 

For instance, the formulation "quite some overhead for getting stopped" could be precised by measuring how long is the overhead so that users can decide whether it is problematic for their use-case or not.

Reply:
Right now, users do not have a choice, because checkpointing is not implemented, so they need to use the stopping variant.

But the paragraph is removed.

We can continue this back and forth, but you really need to accept that I'll continue to push for improvements on the back-end side, because I think it is very important.

In order to make progress with the SM back-end, we need to indeed dive deep and consider what its properties really are. For example, I suggested to consider a variant for PBT where the different workers are run as different instances of the same SM training job. Since workers are started synchronously, this could save overhead.

As long as we do not acknowledge that there is a problem, we will end up not working on alternatives.